edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesEnvironmentOperator.cs

```diff
@@ -37,43 +38,67 @@ public void Stop()
 
         public void Dispose() => this.Stop();
 
-        void StartListPods() =>
+        void StartListPods(CancellationTokenSource shutdownCts) =>
             this.client.ListNamespacedPodWithHttpMessagesAsync(this.deviceNamespace, watch: true)
-                .ContinueWith(this.OnListPodsCompleted);
+                .ContinueWith(this.OnListPodsCompleted, shutdownCts);
 
-        async Task OnListPodsCompleted(Task<HttpOperationResponse<V1PodList>> task)
+        async Task OnListPodsCompleted(Task<HttpOperationResponse<V1PodList>> task, object shutdownCtsObject)
```

Comment:
Why the parameter shutdownCtsObject is an object and not CancellationTokenSource?

Reply:
The [ContinueWith](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.continuewith?view=net-5.0#System_Threading_Tasks_Task_ContinueWith_System_Action_System_Threading_Tasks_Task_System_Object__System_Object_) on line 43 function is defined as `Action<Task,Object>` I'm probably being overly cautious. about the parameter type.